### PR TITLE
Add option to show team responsible for dependencies

### PR DESCRIPTION
**Pull request by MahmudH**
This is to allow shared dependabot responsibilities between different
teams. This will allow teams to be notified by slack when a PR requires
merging.

Co-authored-by: Alan Gabbianelli <alan.gabbianelli@digital.cabinet-office.gov.uk>

Trello: https://trello.com/c/2SYgweYT/1590-3-create-application-ownership-override-to-govuk-dependencies

Screen Shot 2019-12-10 at 12 41 00
Screen Shot 2019-12-10 at 12 41 21
Screen Shot 2019-12-10 at 12 39 52

### CODE REVIEW

File: source/templates/application_template.html.md.erb [hunk 1]
```diff
@@ -27,6 +27,13 @@ GitHub topics: <%= application.topics.map { |topic| link_to(topic, "https://gith
 Owned by team **<%= application.team %>**.
 <% end %>
 
+<% if application.dependencies_team %>
+
+### Dependencies Team
+
+The team responsible to update dependencies is **<%= application.dependencies_team %>**.
```
Comment (issyl0) [inline source/templates/application_template.html.md.erb]:
```suggestion
The team responsible for updating dependencies is **<%= application.dependencies_team %>**.
```

File: data/applications.yml [hunk 2]
```diff
@@ -32,6 +33,7 @@
 - github_repo_name: content-tagger
   type: Publishing apps
   team: "#govuk-platform-health"
+  dependencies_team: "#govuk-platform-health"
```
Comment (thomasleese) [inline data/applications.yml]:
```suggestion
```

We don't need this as the team is already `#govuk-platform-health`.

File: data/applications.yml [hunk 3]
```diff
@@ -329,6 +341,7 @@
 - github_repo_name: mapit
   type: APIs
   team: "#govuk-platform-health"
+  dependencies_team: "#govuk-platform-health"
```
Comment (thomasleese) [inline data/applications.yml]:
```suggestion
```

The team is also set to `#govuk-platform-health`.

File: data/applications.yml [hunk 4]
```diff
@@ -406,6 +421,7 @@
 - github_repo_name: router
   type: Supporting apps
   team: "#govuk-platform-health"
+  dependencies_team: "#govuk-platform-health"
```
Comment (thomasleese) [inline data/applications.yml]:
```suggestion
```

The team is already set to `#govuk-platform-health`.

File: data/applications.yml [hunk 5]
```diff
@@ -640,6 +663,7 @@
   type: Licensing apps
   production_hosted_on: aws
   team: "#govuk-licensing"
+  dependencies_team: "#govuk-licensing"
```
Comment (thomasleese) [inline data/applications.yml]:
```suggestion
```

The team is already set to `#govuk-licensing`.

File: data/applications.yml [hunk 6]
```diff
@@ -376,6 +389,7 @@
 - github_repo_name: search-admin
   type: Supporting apps
   team: "#govuk-searchandnav"
+  dependencies_team: "#govuk-searchandnav"
```
Comment (thomasleese) [inline data/applications.yml]:
```suggestion
```

The team is already set to `#govuk-searchandnav`.

**Pull request merged**